src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java

```diff
@@ -280,21 +304,32 @@ private RetryUtils.RetryConfig getRetryConfig() {
     public void uploadCoreDeviceCAs(String thingName) throws CertificateEncodingException, KeyStoreException,
             IOException {
         List<String> certificatePemList = getCACertificates();
+        List<Class> retryAbleExceptions = Arrays.asList(ThrottlingException.class, InternalServerException.class,
+                AccessDeniedException.class);
+
+        RetryUtils.RetryConfig retryConfig  = RetryUtils.RetryConfig.builder()
+                .initialRetryInterval(Duration.ofSeconds(3)).maxAttempt(Integer.MAX_VALUE)
+                .retryableExceptions(retryAbleExceptions)
+                .build();
 
         PutCertificateAuthoritiesRequest request =
                 PutCertificateAuthoritiesRequest.builder().coreDeviceThingName(thingName)
                         .coreDeviceCertificates(certificatePemList).build();
+
         try {
-            RetryUtils.runWithRetry(getRetryConfig(),
+            RetryUtils.runWithRetry(retryConfig,
                     () -> clientFactory.getGreengrassV2DataClient().putCertificateAuthorities(request),
                     "put-core-ca-certificate", logger);
         } catch (InterruptedException e) {
             logger.atInfo().log("Put core CA certificates got interrupted");
             // interrupt the current thread so that higher-level interrupt handlers can take care of it
             Thread.currentThread().interrupt();
         } catch (Exception e) {
-            throw new CloudServiceInteractionException("Failed to put core CA certificates to cloud. Check that the "
-                    + "core device's IoT policy grants the greengrass:PutCertificateAuthorities permission.", e);
+            String errMsg = "Failed to put core CA certificates to cloud. Check that the "
```

Comment:
why is this logging here now instead of throwing as before? The caller no doesn't know if it succeeds.